cli/proc_state.rs

```diff
@@ -299,14 +300,26 @@ impl ProcState {
     let maybe_locker = as_maybe_locker(self.lockfile.clone());
     let maybe_imports = self.get_maybe_imports();
     let node_resolver = NodeEsmResolver::new(
-      self.maybe_import_map.as_ref().map(ImportMapResolver::new),
+      self.maybe_import_map.clone().map(ImportMapResolver::new),
     );
-    let import_map_resolver =
-      self.maybe_import_map.as_ref().map(ImportMapResolver::new);
+    let maybe_import_map_resolver =
+      self.maybe_import_map.clone().map(ImportMapResolver::new);
+    let maybe_jsx_resolver = self
+      .maybe_config_file
+      .as_ref()
+      .map(|cf| {
+        cf.to_maybe_jsx_import_source_module()
+          .map(|im| JsxResolver::new(im, maybe_import_map_resolver.clone()))
+      })
+      .flatten();
     let maybe_resolver = if self.flags.compat {
       Some(node_resolver.as_resolver())
+    } else if maybe_jsx_resolver.is_some() {
+      maybe_jsx_resolver.as_ref().map(|jr| jr.as_resolver())
```

Comment:
Reading this code makes me feel like this can't be used with import maps, but that's not the case because the JSX resolver takes an import map resolver and handles all that. Maybe we should add a comment here that says that. I also wonder if we should extract out the code that gets the resolver to some other common place for reuse.

Reply:
For the latter part, feels like something we should address in #12684.